### PR TITLE
Port lodash methods to private functions. Issue #69

### PR DESCRIPTION
**Pull request by dtnhinh**
- [x] Related GitHub issue(s) linked in PR description
- [x] Destination branch merged, built and tested with your changes
- [x] Code formatted and follows best practices and patterns
- [x] Code builds cleanly (no _additional_ warnings or errors)
- [x] Manually tested
- [x] Automated tests are passing
- [x] No _decreases_ in automated test coverage
- [x] Documentation updated (readme, docs, comments, etc.)
- [x] Localization: No hard-coded error messages in code files (_minimally_ in string constants)

### CODE REVIEW

File: src/utilities/core-utils.ts [hunk 1]
```diff
@@ -56,7 +107,12 @@ const _numericEnumToPojo = (enumObject: any): {} => {
     return pojo;
 };
 
-const _objectToArray = (object: any): any[] => {
+const memoize = <T extends (...args: any[]) => any>(
```
Comment (brandongregoryscott) [inline src/utilities/core-utils.ts:110]: Nitpick - alphabetization

File: src/utilities/core-utils.ts [hunk 2]
```diff
@@ -129,27 +225,41 @@ const _timer = (name: string) => {
...
+
+/**
+ * Invokes the iteratee function n times, returning an array of the results of each invocation. The iteratee
+ * is invoked with one argument; (index).
+ *
+ * @param n The number of times to invoke iteratee.
+ * @param iteratee The function invoked per iteration.
+ * @return Returns the array of results.
+ */
+const times = <TResult>(
```
Comment (brandongregoryscott) [inline src/utilities/core-utils.ts:237]: Oh, sweet! Didn't even realize this was something we had available. Kinda the same idea that I had with a recent function export from the testing package to loop a test body a certain number of times.

**Review by brandongregoryscott — APPROVED** [review]
Approving, thanks for sweeping through these @nhinhdao! Just one alphabetization error I found. And a final thought/question, but I _think_ this is the case based on what I was seeing:

Without copy/pasting the JSDoc comments provided from the lodash functions, they will not be forwarded automatically in the transpiled JS/type definition files.

Just wanted to double check to make sure before we go ahead with copying these over each wrapper function - it'd be great to get those for free if we can.

File: src/utilities/core-utils.ts (hunk 1 above)
Comment (dtnhinh) [reply src/utilities/core-utils.ts:110]: Yeah, I tried removing the comments I copied over and it wouldn't show anymore even with the direct export. I think it doesn't hurt to just copy over so I just do them all like that

**Pull request merged**